### PR TITLE
Add deprecated warning test to the multi-objective sampler test file

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Followup as mentioned in https://github.com/optuna/optuna/pull/3569#discussion_r876621912.

## Description of the changes

Add a test to check the deprecated warning in the multi-objective module.

### CODE REVIEW

File: tests/multi_objective_tests/samplers_tests/test_samplers.py [hunk 1]
```diff
@@ -81,6 +81,21 @@ def test_random_mo_sampler_reseed_rng() -> None:
...
 
 
+@pytest.mark.parametrize(
+    "sampler_class",
+    [
+        optuna.multi_objective.samplers.RandomMultiObjectiveSampler,
+        optuna.multi_objective.samplers.NSGAIIMultiObjectiveSampler,
+        optuna.multi_objective.samplers.MOTPEMultiObjectiveSampler,
+    ],
+)
```
Comment (nzw0301) [inline tests/multi_objective_tests/samplers_tests/test_samplers.py:91]: Alternatively, we might add `MOTPEMultiObjectiveSampler` to https://github.com/optuna/optuna/blob/af2ba57dc034660fe77979959a5313b08c62bd74/tests/multi_objective_tests/samplers_tests/test_samplers.py#L16-L22.

**Comment by nzw0301** [comment]
Should I add similar tests for the other deprecated method/class in `optuna/multi_objective`? In my understanding, we do not have such tests for this module:

```bash
# on master branch
$ /optuna/tests/multi_objective_tests> grep -i "FutureWarning" -r .
./samplers_tests/test_motpe.py:@pytest.mark.filterwarnings("ignore::FutureWarning")
./samplers_tests/test_motpe.py:@pytest.mark.filterwarnings("ignore::FutureWarning")
./samplers_tests/test_motpe.py:@pytest.mark.filterwarnings("ignore::FutureWarning")
./samplers_tests/test_motpe.py:@pytest.mark.filterwarnings("ignore::FutureWarning")
```

**Comment by toshihikoyanase** [comment]
@HideakiImamura Could you review this PR as an author of [this comment](https://github.com/optuna/optuna/pull/3569#discussion_r876621912)? Please feel free to remove the assignment if you are busy.

File: tests/multi_objective_tests/samplers_tests/test_samplers.py (hunk 1 above)
Comment (toshihikoyanase) [reply tests/multi_objective_tests/samplers_tests/test_samplers.py:91]: Not a strong opinion, but I think we do not need to improve the test coverage of the deprecated sampler. So, I prefer the current change.

**Review by toshihikoyanase — APPROVED** [review]
LGTM. Thank you!

> Should I add similar tests for the other deprecated method/class in `optuna/multi_objective`?

I found that the `deprecated` methods/classes in the other parts (i.e., `optuna/distribution.py` and `optuna/trial/*.py`) also lack corresponding test cases. I don't think it is necessary to implement the tests unless we have specific reason.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**